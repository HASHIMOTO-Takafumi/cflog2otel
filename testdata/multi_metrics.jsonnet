local cel = std.native('cel');

{
  otel: {
    endpoint: 'http://localhost:4317/',
    gzip: true,
  },
  resource_attributes: [
    {
      key: 'service.name',
      value: 'Amazon CloudFront',
    },
    {
      key: 'aws.cloudfront.distribution_id',
      value: cel('cloudfront.distributionId'),
    },
  ],
  scope: {
    name: 'test',
  },
  metrics: [
    {
      name: 'http.server.requests',
      description: 'The number of HTTP requests',
      type: 'Count',
      attributes: [
        {
          key: 'http.status_code',
          value: cel('log.scStatusCategory'),
        },
      ],
    },
    {
      name: 'http.server.total_bytes',
      description: 'The total number of bytes sent by the server',
      type: 'Sum',
      unit: 'Byte',
      attributes: [
        {
          key: 'http.status_code',
          value: cel('log.scStatusCategory'),
        },
      ],
      value: cel('double(log.scBytes)'),
      is_monotonic: true,
    },
  ],
}
