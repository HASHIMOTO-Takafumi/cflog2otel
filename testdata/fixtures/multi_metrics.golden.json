{
  "Resource": [
    {
      "Key": "aws.cloudfront.distribution_id",
      "Value": {
        "Type": "STRING",
        "Value": "EMLARXS9EXAMPLE"
      }
    },
    {
      "Key": "service.name",
      "Value": {
        "Type": "STRING",
        "Value": "Amazon CloudFront"
      }
    }
  ],
  "ScopeMetrics": [
    {
      "Scope": {
        "Name": "test",
        "Version": "",
        "SchemaURL": ""
      },
      "Metrics": [
        {
          "Name": "http.server.requests",
          "Description": "The number of HTTP requests",
          "Unit": "",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "2xx"
                    }
                  }
                ],
                "StartTime": "2019-12-01T22:42:00Z",
                "Time": "2019-12-01T22:43:00Z",
                "Value": 3
              },
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-01T22:51:00Z",
                "Time": "2019-12-01T22:52:00Z",
                "Value": 3
              }
            ],
            "Temporality": "DeltaTemporality",
            "IsMonotonic": true
          }
        },
        {
          "Name": "http.server.total_bytes",
          "Description": "The total number of bytes sent by the server",
          "Unit": "Byte",
          "Data": {
            "DataPoints": [
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "2xx"
                    }
                  }
                ],
                "StartTime": "2019-12-01T22:42:00Z",
                "Time": "2019-12-01T22:43:00Z",
                "Value": 1176
              },
              {
                "Attributes": [
                  {
                    "Key": "http.status_code",
                    "Value": {
                      "Type": "STRING",
                      "Value": "5xx"
                    }
                  }
                ],
                "StartTime": "2019-12-01T22:51:00Z",
                "Time": "2019-12-01T22:52:00Z",
                "Value": 2700
              }
            ],
            "Temporality": "DeltaTemporality",
            "IsMonotonic": true
          }
        }
      ]
    }
  ]
}